src/components/Tabs/index.js

```diff
@@ -0,0 +1,223 @@
+    variant: PropTypes.string,
+    accent: PropTypes.string,
+    weight: PropTypes.string,
+    underlineColor: PropTypes.string
+  };
+
+  static defaultProps = {
+    index: -1,
+    headingIndex: -1,
+    onClick: () => {},
+    variant: null,
+    accent: null,
+    weight: null,
+    underlineColor: null
+  };
+
+  state = {
+    isTabsContainerOverflows: false
+  };
+
+  componentDidMount() {
+    this.updateIsOverflowsState();
+    window.addEventListener("resize", this.updateIsOverflowsState); // eslint-disable-line
+  }
+
+  componentDidUpdate() {
+    if (this.state.isTabsContainerOverflows) {
+      const {
+        offsetWidth: contentWidth,
+        scrollLeft: contentScrollLeft
+      } = this.content;
+      const { index } = this.props;
+      const {
+        offsetWidth: activeItemWidth = 0,
+        offsetLeft: activeItemLeft = 0
+      } = this.itemRefs[index];
+
+      if (activeItemWidth + activeItemLeft > contentWidth + contentScrollLeft) {
+        this.content.scrollLeft =
+          activeItemWidth +
+          activeItemLeft -
+          contentWidth +
+          parseInt(padding, 10);
+      } else if (activeItemLeft < contentScrollLeft) {
+        this.content.scrollLeft = activeItemLeft - parseInt(padding, 10);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateIsOverflowsState); //eslint-disable-line
+  }
+
+  updateIsOverflowsState = () => {
+    const isOverflows = checkIfOverflows(this.content);
+    if (this.state.isTabsContainerOverflows !== isOverflows) {
+      this.setState({
+        isTabsContainerOverflows: isOverflows
+      });
+    }
+  };
+
+  itemRefs = [];
+
+  renderItem = ({ itemIndex, item, ...itemprops }) => {
+    const { index, onClick, underlineColor } = this.props;
+    return (
+      <TabItemButton
+        isActive={itemIndex === index}
+        underlineColor={underlineColor}
+        onClick={() => onClick(itemIndex)}
+        innerRef={ref => {
+          this.itemRefs[itemIndex] = ref;
+        }}
+        dataIndex={itemIndex}
+      >
+        <Text
+          responsiveSize={{ small: "kilo", large: "giga" }}
+          primary
+          {...itemprops}
+        >
+          {item}
+        </Text>
+      </TabItemButton>
+    );
+  };
+
+  render() {
+    const { items, headingIndex, variant, accent, weight } = this.props;
+
+    const areItemsValidKeys =
+      items &&
+      items.length > 0 &&
+      (items[0].constructor === String || items[0].constructor === Number);
+
+    return (
+      <Container areTabsOverflow={this.state.isTabsContainerOverflows}>
+        <Content
+          innerRef={ref => {
+            this.content = ref;
+          }}
+          areTabsOverflow={this.state.isTabsContainerOverflows}
+        >
+          {items.map((item, itemIndex) => (
+            <Item key={areItemsValidKeys ? item : itemIndex}>
+              {itemIndex === headingIndex ? (
+                <h2 style={{ margin: 0 }}>
```

Comment:
I don't believe we want to hardcode an `h2` here. Also, can you please make this more composable, rather than using `this.renderItem`?

Reply:
Done. Thanks.